unstructured/partition/auto.py

```diff
@@ -141,12 +141,12 @@ def partition(
     encoding: Optional[str] = None,
     paragraph_grouper: Optional[Callable[[str], str]] | Literal[False] = None,
     headers: dict[str, str] = {},
-    skip_infer_table_types: list[str] = [],
+    skip_infer_table_types: list[str] = ["pdf", "jpg", "png", "xls", "xlsx", "heic"],
```

Comment:
I recommend we only include PDF and image types in this list, not `xls` or `xlsx`.

### Rationale
The "infer_table" behavior is historically somewhat confused when it comes to non-PDF/Image file-types.

The `infer_table_structure` argument (predecessor to `skip_infer_table_types`) controlled two things and we often want those two controlled separately for non-PDF/Image types.

1. For PDF/Image file-types, `infer_table_types` controlled partitioning "expense", both money and time, by controlling whether the extra compute of using a table-inference model was engaged. This was important (and still is) for PDF/Image because doing this can take a lot more CPU and elapsed-time.
2. It _also_ determined whether `.metadata.text_as_html` was added to `Table` elements. For PDF/Image this was obvious enough because if you didn't infer tables there was no `Table` element to put the `.metadata.text_as_html` on.
3. However, for non-PDF/Image file-types, no inference was ever required or used to detect tables. So `Table` elements always appear for those file-types when they occur in the document.
4. In that non-PDF/Image case, `infer_table_structure` came to mean: "include `.metadata.text_as_html` on `Table` elements".
### My POV
- In my opinion, any "infer_table" option, regardless of how it is specified, should be limited to PDF/Image types.
- In addition, we can have a `suppress_metadata_fields: list[str] = []` parameter that is applicable to all partitioners and handled in post-partitioning processing (implemented in one of the decorators), and that removes any metadata the user prefers not to appear in the payload.
- In the meantime, I believe only PDF/Image partitioners are actually sensitive to the value of `skip_infer_table_types`, so probably not a big deal, but still, better to remove `xls` and `xlsx` from the list I think.

Reply:
Agree with all this.

Reply:
Updated!